### DISCUSSION TITLE
Injecting custom Clock to jobrunr

### DISCUSSION THREAD

**Discussion by dkubicki** [jobrunr/jobrunr#935] [General]
Hello community,

I have a question about injecting a custom clock into JobRunr.
Why do I want to do this? It's because I want to perform integration tests, test my jobs and configurations, and have the capability to manipulate time.

Any advice on how to accomplish this would be greatly appreciated. Thank you! :)

**Comment by rdehuyss** [comment]
Personally, I find the use of custom clock's not so readable. That's why I use Mockito to freeze Instant.now() to a timestamp of my choice.

**Reply by dkubicki** [reply]
Thanks for your response. After walking through the Jobrunr configuration (including autoconfiguration), I don't see that any clock is injected. This means that even if I want to use my approach I'm not able to do it properly?
